db/migrate/20210103181832_connect_ad_with_city.rb

```diff
@@ -0,0 +1,13 @@
+class ConnectAdWithCity < ActiveRecord::Migration[6.1]
+  def change
+    ActiveRecord::Base.transaction do
+      Rake::Task['db:seed_locations'].invoke
+      rename_column :ads, :city, :city_zdel
+      add_reference :ads, :city, foreign_key: true, index: true
```

Comment:
```suggestion
      add_reference :ads, :city, foreign_key: true, index: true, null: false
```

Reply:
This won't work immediately, because we have ads in our database and, when we add `city_id` reference, none of the ads will have `city_id` not null.

We can do:

```rb
      add_reference :ads, :city, foreign_key: true, index: true
      # ... do the data migration
      change_column_null :ads, :city_id, false
```

Reply:
True

Reply:
I think we should use this as opportunity to clean up data before we have too many ads